pangea-sdk/service/ai_guard/api.go

```diff
@@ -255,18 +263,37 @@ type CodeDetectionResult struct {
 	Action   string `json:"action"`
 }
 
+type SingleEntityResult struct {
+	Action   string   `json:"action"`   // The action taken by this Detector
+	Entities []string `json:"entities"` // Detected entities
+}
+
+type Classification struct {
+	Category   string  `json:"category"`
+	Confidence float64 `json:"confidence"`
+}
+
+type ClassificationResult struct {
+	Action          string           `json:"action"` // The action taken by this Detector
+	Classifications []Classification `json:"classifications"`
+}
+
 type TextGuardDetector[T any] struct {
 	Detected bool `json:"detected"`
 	Data     *T   `json:"data,omitempty"`
 }
 
 type TextGuardDetectors struct {
 	PromptInjection      *TextGuardDetector[PromptInjectionResult]   `json:"prompt_injection,omitempty"`
+	Gibberish            *TextGuardDetector[ClassificationResult]    `json:"gibberish,omitempty"`
+	Sentiment            *TextGuardDetector[ClassificationResult]    `json:"sentiment,omitempty"`
+	SelfHarm             *TextGuardDetector[ClassificationResult]    `json:"selfharm,omitempty"`
 	PiiEntity            *TextGuardDetector[PiiEntityResult]         `json:"pii_entity,omitempty"`
 	MaliciousEntity      *TextGuardDetector[MaliciousEntityResult]   `json:"malicious_entity,omitempty"`
+	CustomEntity         *TextGuardDetector[RedactEntityResult]      `json:"custom_entity,omitempty"`
 	SecretsDetection     *TextGuardDetector[SecretsEntityResult]     `json:"secrets_detection,omitempty"`
-	ProfanityAndToxicity *TextGuardDetector[any]                     `json:"profanity_and_toxicity,omitempty"`
-	CustomEntity         *TextGuardDetector[any]                     `json:"custom_entity,omitempty"`
```

Comment:
Should we add this to changelog as changed or removed?

Reply:
I only moved them around to match the ordering on the API docs page.